doc/user_guide/io.rst

```diff
@@ -1027,11 +1033,279 @@ HDF5 file. All other keyword arguments will be passed to
+``metadata.Acquisition_instrument.SEM.beam_current`` would need to change to 
+``metadata.Acquisition_instrument.SEM.beam_current.value`` for example. 
+As this will likely break some specific hyperspy functions
+It's recommended that a "mapping" dictionary is passed to the load or
+file_reader to translate stored original_metadata to 
+hyperspy signal metadata.
+For example to map a hdf5 entry to hyperspy metadata.
+
+.. code-block:: python
+
+   >>> nexus_mapping = {"entry.instrument.beamline.DCM.dcm_energy":("Acquisition_instrument.TEM.beam_energy",None)}
+   >>> a = hs.load("test.nxs",mapping=nexus_mapping)
+      
+Hyperspy metadata is stored with "original_metadata" in the file
+Hyperspy metadata is not automatically restored when a signal is loaded
+from a Nexus file. A mapping dictionary is the preferred method to map
+file metadata to hyperspy metdata.   
+
+Inspecting
+^^^^^^^^^^
+Looking in a Nexus file for specific metadatda is often useful - .e.g to find
+what position a specific stage was at: 
+
+    >>> from hyperspy.io_plugins.nexus import get_metadata_in_file
+    >>> results=get_metadata_in_file("sample.hdf5",search_keys=["stage1_z"])
+    {'entry': {'instrument': {'scannables': {'stage1': {'stage1_z': {'value': -9.871700000000002,
+    'attrs': {'gda_field_name': 'stage1_z',
+    'local_name': 'stage1.stage1_z',
+    'target': '/entry/instrument/scannables/stage1/stage1_z',
+    'units': 'mm'}}}}}}} 
+    
+You can also inspect the file to see what datasets are stored in the file
+    
+    >>> from hyperspy.io_plugins.nexus import get_data_in_file
+    >>> results=get_metadata_in_file("sample.hdf5",search_keys=["stage1_z"])
+    NXdata found
+    /entry/xsp3_addetector
+    /entry/xsp3_addetector_total
+    HDF datasets found
+    /entry/solstice_scan/keys/uniqueKeys
+    /entry/solstice_scan/scan_shape
+    Out[3]:
+    (['/entry/xsp3_addetector', '/entry/xsp3_addetector_total'],
+     ['/entry/solstice_scan/keys/uniqueKeys', '/entry/solstice_scan/scan_shape'])    
+    
+
+Reading
+^^^^^^^
+Nexus files can contain multiple datasets within the same file but the
+ordering of datasets can vary depending on the setup of an experiment or
+processing step when the data was collected.
+For example in experiment 1 Fe,Ca,P,Pb were collected in experiment 2
+Ca,P,K,Fe,Pb were collected. HyperSpy supports reading in one or more datasets and returns a list
+of signals but in this example case the indexing is different.
+To control which data or metadata is loaded and in what order
+some additional loading arguments are provided.
+
+Extra loading arguments
++++++++++++++++++++++++
+- ``dset_search_keys`` : ``str`` or ``list`` - Absolute path(s) or substring(s) used to find one or more datasets. (default is ``None`` all Nexus Datasets will be read)
+- ``meta_search_keys`` : ``str`` or ``list`` of strings - Absolute path(s) or substring(s) to use to find one or more datasets. (default is ``None`` all Nexus Metadata will be read)
```

Comment:
Similar as above: would ``metadata_keys`` be better?

Reply:
dset was used in the usid plugin and docs so was trying to be consistent - but can change

Reply:
Changed to dataset_keys and metadata_keys as suggested.